### DISCUSSION TITLE
Background Opacity Setting Affecting Neovim Background

### DISCUSSION THREAD

**Discussion by timoclsn** [ghostty-org/ghostty#3579] [Feature Requests, Ideas] [answered]
I'm using Ghostty on macOS und i like my terminal to have some background opacity.

I am using this config:

```
background-opacity = 0.9
background-blur-radius = 20
```

I'm using Neovim with the tokyonight theme. I have set transparency to false for tokynight in Neovim.

But it seems like the Ghostty background opacity setting is also affecting the Neovim background. In all the other terminal emulators i have tested (Wezterm, Alacritty) this is not the case. Which i like!

Is there a way to get this behaviour in Ghostty? Solid BG in Neovim but general transparent terminal?

(If i wanted I always could make my Neovim background transparent and have the terminal setting take affect this way)

**Comment by jmckillican** [comment]
I would look to your nvim configs, by default for my neovim window was a solid background color while my ghostty config was set to 80% until I added these lines to my init.lua

```
vim.cmd([[
  augroup TransparentBackground
  autocmd!
  autocmd ColorScheme * highlight Normal ctermbg=none guibg=none
  autocmd ColorScheme * highlight NonText ctermbg=none guibg=none
  augroup END
]])
```

**Reply by timoclsn** [reply]
But what you described first ist exactly what I want:

- Ghostty Opacity 0.9
- Neovim solid background

I think [this](https://github.com/ghostty-org/ghostty/discussions/3573) could be the "issue" as I have Ghostty and Neovim set to tokyonight.

**Comment by timoclsn** [comment]
I now have worked around it by overriding the terminal background with a sliiiightly other background color so that it doesn't match my editor theme.

```
theme = tokyonight
# Slightly changed background color so that neovim background doesn't get transparent
background = 1A1B25 
```

But would be nice to know if this is a bug or intended behaviour.

**Comment by gigamonster256** [comment]
You can easily recreate the behavior using ghostty itself, specifically when ruinning `ghostty +list-themes` with a background-opacity < 1, it will show a transparent background when previewing your current theme, but be opaque when previewing any other theme.

**Comment by 00-kat** [comment]
I don't think this issue is intentional; someone also found this on the [Ghostty Discord] later, and one of the maintainers was able to figure out why it's broken.  Relevant Discord thread: https://discord.com/channels/1005603569187160125/1331382035507314761.

[Ghostty Discord]: https://discord.gg/ghostty

**Comment by 00-kat** [comment]
#5001 is a duplicate of this and the OP there notes that this has been fixed now, could someone try it again to confirm it?

**Reply by timoclsn** [reply, marked as answer]
Yes, I checked and this has been fixed in 1.1.0 ✅
